File: fake-store-quarkus/src/main/java/org/arielzarate/infrastructure/persistence/entity/ProductEntity.java
package org.arielzarate.infrastructure.persistence.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;


@Entity
@Table(name = "products")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ProductEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private Double price;

    @Column(columnDefinition = "TEXT")
    private String description;

    private String category;

    private String image;

    @Embedded
    private RatingEmbed rating;

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    @Embeddable
    public static class RatingEmbed {
        private Double rate;
        private Integer count;
    }
}

File: api-fake-hexagonal/src/main/kotlin/com/arielZarate/fake_api_hexagonal/infraestructure/persistence/model/ProductEntity.kt
package com.arielZarate.fake_api_hexagonal.infraestructure.persistence.model

import jakarta.persistence.*


@Entity
@Table(name = "product")
class ProductEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int = 0,
    var title: String,
    var price: Double,
    @Column(length = 700)
    var description: String,
    var category: String,
    var image: String,
    @Embedded
    var rating: RatingEntity
) {


}


